be/src/storage/compaction_context.h

```diff
@@ -0,0 +1,48 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+#include <memory>
+#include <set>
+#include <vector>
+
+#include "storage/rowset/rowset.h"
+#include "storage/tablet.h"
+
+namespace starrocks {
+
+class RowsetReleaseGuard;
+class CompactionTask;
+
+#ifndef LEVEL_NUMBER
+#define LEVEL_NUMBER 3
+#endif
+
+const double COMPACTION_SCORE_THRESHOLD = 0.999;
+
+struct RowsetComparator {
+    bool operator()(const Rowset* left, const Rowset* right) const {
+        return left->start_version() < right->start_version() && left->end_version() < right->start_version();
```

Comment:
Comparator needs proper requirements

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/n4367.html

If versions will never overlap, the simple version should be left->start < right->start
If they can overlap, should be left->start < right->start || (left->start == right->start && left->end < right->end)

Reply:
fixed. PTAL